src/main/java/com/ita/if103java/ims/dao/impl/DashboardDaoImpl.java

```diff
@@ -4,111 +4,202 @@
 import com.ita.if103java.ims.dto.PopularityListDto;
 import com.ita.if103java.ims.dto.RefillListDto;
 import com.ita.if103java.ims.dto.WarehouseLoadDto;
+import com.ita.if103java.ims.dto.WarehousePremiumStructDto;
+import com.ita.if103java.ims.entity.DateType;
+import com.ita.if103java.ims.entity.PopType;
 import com.ita.if103java.ims.exception.CRUDException;
 import com.ita.if103java.ims.mapper.jdbc.PopularityListRowMapper;
 import com.ita.if103java.ims.mapper.jdbc.RefillListRowMapper;
 import com.ita.if103java.ims.mapper.jdbc.WarehouseLoadRowMapper;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import com.ita.if103java.ims.mapper.jdbc.WarehousePremiumLoadRowMapper;
+import com.ita.if103java.ims.mapper.jdbc.WarehousePremiumStructRowMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.dao.DataAccessException;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Repository;
-
-
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.ArrayList;
 import java.util.List;
 
+import static com.ita.if103java.ims.entity.PopType.TOP;
+
 @Repository
 public class DashboardDaoImpl implements DashboardDao {
     private final static Logger LOGGER = LoggerFactory.getLogger(DashboardDaoImpl.class);
     private WarehouseLoadRowMapper warehouseLoadRowMapper;
     private PopularityListRowMapper popularityListRowMapper;
     private RefillListRowMapper refillListRowMapper;
+    private WarehousePremiumStructRowMapper warehousePremiumLoadMapper;
+    private WarehousePremiumLoadRowMapper warehousePremiumLoadRowMapper;
     private JdbcTemplate jdbcTemplate;
 
     @Autowired
     public DashboardDaoImpl(WarehouseLoadRowMapper warehouseLoadRowMapper,
                             RefillListRowMapper refillListRowMapper,
                             PopularityListRowMapper popularityListRowMapper,
+                            WarehousePremiumStructRowMapper warehousePremiumLoadMapper,
+                            WarehousePremiumLoadRowMapper warehousePremiumLoadRowMapper,
                             JdbcTemplate jdbcTemplate) {
+        this.warehousePremiumLoadMapper = warehousePremiumLoadMapper;
         this.refillListRowMapper = refillListRowMapper;
         this.popularityListRowMapper = popularityListRowMapper;
         this.warehouseLoadRowMapper = warehouseLoadRowMapper;
+        this.warehousePremiumLoadRowMapper = warehousePremiumLoadRowMapper;
         this.jdbcTemplate = jdbcTemplate;
     }
     @Override
     public List<WarehouseLoadDto> findWarehouseWidgets() {
         try {
             return jdbcTemplate.query(Queries.SQL_FIND_WAREHOUSE_WIDGETS, warehouseLoadRowMapper);
         }catch (DataAccessException e) {
-            throw crudException("WarehouseWidget",e.toString());
+            throw new CRUDException("DashboardDaoImpl failed to find WarehouseWidget",e);
```

Comment:
...to find Warehouse load instead of Warehouse widget.

Widget is a part of the dashboard, so better to call methods by mentioning what we are going to select